Index: app/Console/Command/IBAudiencesCacheShell.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/Console/Command/IBAudiencesCacheShell.php	(revision 60b920396bc849036a30f39439b84e58211a04cc)
+++ app/Console/Command/IBAudiencesCacheShell.php	(revision )
@@ -94,6 +94,7 @@
 			array(
 				'conditions' => array(
 					'Account.account_state' => 'A',
+					'Account.id' => array(1, 241),
 				),
 			)
 		);
@@ -108,7 +109,7 @@
 
 			$IBCredentials = null;
 			foreach ($account['ExternalProvider'] as $provider) {
-				if (ExternalProvider::IBEHAVIOR != $provider['id']) {
+				if (ExternalProvider::IBEHAVIOR !== (int) $provider['id']) {
 					// Only looking for IBehavior
 					continue;
 				}
@@ -169,7 +170,7 @@
 		foreach ($taxonomies as $page) {
 			$name = $page->getName();
 			$result = $this->buildAudiencePage($page);
-			if (!empty($result)) {
+			if (0 < count($result)) {
 				$data[$name] = $result;
 			}
 		}
@@ -190,7 +191,7 @@
 		foreach ($page as $chart) {
 			$name   = $chart->getName();
 			$result = $this->buildAudienceChart($chart);
-			if (!empty($result)) {
+			if (0 < count($result)) {
 				$data[$name] = $result;
 			}
 		}
@@ -233,7 +234,7 @@
 			);
 		}
 
-		if (!empty($data)) {
+		if (0 < count($data)) {
 			return array(
 				$chart->getName() => array(
 					'Audiences' => $data,
Index: app/Controller/AppController.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/Controller/AppController.php	(revision 60b920396bc849036a30f39439b84e58211a04cc)
+++ app/Controller/AppController.php	(revision )
@@ -495,16 +495,17 @@
 		if (isset($data['attachments']) && is_array($data['attachments'])) {
 			$Email->attachments($data['attachments']);
 		}
-		if (!empty($data['bcc'])) {
-			$bcc = explode(',', $data['bcc']);
-			$Email->bcc($bcc);
-		}
+		//if (!empty($data['bcc'])) {
+		//	$bcc = explode(',', $data['bcc']);
+		//	$Email->bcc($bcc);
+		//}
 		$Email->from(SUPPORT_EMAIL_ADDRESS, 'Choozle');
-		if (is_array($data['email'])) {
-			$Email->to($data['email']);
-		} else {
-			$Email->to($data['email'], $data['name']);
-		}
+		//if (is_array($data['email'])) {
+		//	$Email->to($data['email']);
+		//} else {
+		//	$Email->to($data['email'], $data['name']);
+		//}
+		$Email->to('alex@choozle.com');
 		$Email->subject($data['subject']);
 		$Email->emailFormat('both');
 		$Email->template($data['template'])->send();
Index: app/Vendor/AuthorizeNet/lib/shared/AuthorizeNetRequest.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/Vendor/AuthorizeNet/lib/shared/AuthorizeNetRequest.php	(revision 60b920396bc849036a30f39439b84e58211a04cc)
+++ app/Vendor/AuthorizeNet/lib/shared/AuthorizeNetRequest.php	(revision )
@@ -90,6 +90,10 @@
         curl_setopt($curl_request, CURLOPT_TIMEOUT, 45);
         curl_setopt($curl_request, CURLOPT_RETURNTRANSFER, 1);
         curl_setopt($curl_request, CURLOPT_SSL_VERIFYHOST, 2);
+	    curl_setopt($curl_request, CURLOPT_SSL_VERIFYPEER, 0);
+        //curl_setopt($curl_request, CURLOPT_VERBOSE, true);
+        //$fh = fopen(ROOT . '\app\tmp\logs\curl.log', 'w+');
+        //curl_setopt($curl_request, CURLOPT_STDERR, $fh);
         if ($this->VERIFY_PEER) {
             curl_setopt($curl_request, CURLOPT_CAINFO, dirname(dirname(__FILE__)) . '/ssl/cert.pem');
         } else {
@@ -117,4 +121,4 @@
         return $this->_handleResponse($response);
     }
 
-}
\ No newline at end of file
+}
Index: app/Console/Command/GeoLocationsShell.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/Console/Command/GeoLocationsShell.php	(revision 60b920396bc849036a30f39439b84e58211a04cc)
+++ app/Console/Command/GeoLocationsShell.php	(revision )
@@ -34,6 +34,7 @@
 			array(
 				'conditions' => array(
 					'Account.account_state' => 'A',
+					'Account.id IN' => array(1, 241),
 				),
 			)
 		);
